mesa_frames/concrete/polars/agentset.py

```
@@ -120,20 +121,28 @@ def add(
         """
         obj = self._get_obj(inplace)
         if isinstance(agents, pl.DataFrame):
-            if "unique_id" not in agents.columns:
-                raise KeyError("DataFrame must have a unique_id column.")
+            if "unique_id" in agents.columns:
+                warnings.warn("Dataframe should not have a unique_id index/column. It will be ignored.", DeprecationWarning)
             new_agents = agents
         elif isinstance(agents, dict):
-            if "unique_id" not in agents:
-                raise KeyError("Dictionary must have a unique_id key.")
+            if "unique_id" in agents:
+                warnings.warn("Dictionary should not have a unique_id key. It will be ignored.", DeprecationWarning)
             new_agents = pl.DataFrame(agents)
         else:
-            if len(agents) != len(obj._agents.columns):
+            if len(agents) not in {len(obj._agents.columns), len(obj._agents.columns) + 1}:
                 raise ValueError(
                     "Length of data must match the number of columns in the AgentSet if being added as a Collection."
                 )
+            if len(agents) == len(obj._agents.columns):
+                warnings.warn("Length of data should have the number of columns in the AgentSet," +
+                    "we suppose the first element is the unique_id. It will be ignored.", DeprecationWarning)
             new_agents = pl.DataFrame([agents], schema=obj._agents.schema)
 
```

I think it's better to raise an Exception to be more explicit about it (from the [Zen of Python](https://peps.python.org/pep-0020/), explicit is better than implicit). We only have a pre-stable release anyway so we can afford to make "drastic" changes.